### ISSUE TITLE
[Feature]: Compatibility with currently stable Neovim 0.9.x

### ISSUE THREAD

**Issue by rudionrails** [Bekaboo/dropbar.nvim#107]
### Problem

The plugin requires neovim 0.10.0-dev as minimum version. Seems that it is tapping into an API hat is only available from this version onwards. However, would be nice to establish compatibility with the current stable neovim version out there. Any chance this can be done?

### Expected behavior

possibility to make it work with currently stable neovim 0.9.x version

**Comment by FormalSnake** [comment]
I have been waiting for 0.10 to release for a while, since I rely on neovim too much to go into a dev branch, and thus I would love to see this too.

**Comment by Bekaboo** [comment]
Hi, thanks for coming up with the issue.

> Any chance this can be done?

Yes but it is not in my todo list. 0.10.0 will come stable eventually so I have no insterest in maintaining compatibility with 0.9.

Chances are that someone could send a PR for 0.9, in which case I'll open a new `0.9` branch for users on nvim 0.9. The branch will not receive frequent updates though, because I'm on 0.10 and testing for nvim 0.9 will dramatically increase my workload to maintain this plugin.

**Comment by rudionrails** [comment]
Alrighty. Any hint on what the v0.10 interfaces are that you the plugin depends on?

**Comment by Bekaboo** [comment]
> Any hint on what the v0.10 interfaces are that you the plugin depends on?

Here is an (incomplete) list of differences what I can recall:

1. `vim.iter` module does not exist in `0.9`
2. `vim.loop` (0.9) -> `vim.uv` (0.10)
3. `vim.lsp.get_active_clients` -> `vim.lsp.get_clients`

**Comment by rudionrails** [comment]
OK, I looked into this a little. There are few additional v0.10 specific functions being used. I looked into https://github.com/mhanberg/iter.nvim and tried to find other polyfills (instead of needing to modify the code here). I don't think it is worth it for now (just look at the iter polyfill implementation) and v0.10 seems around the corner. Going for https://github.com/utilyre/barbecue.nvim in the meantime. Will close the issue. Thanks all
